face-detection-explainer.md

```diff
@@ -0,0 +1,481 @@
+
+
+## Performance
+
+Face detection, using the proposed Javascript API, was compared to several other alternatives in power usage.
+The results were normalized against base case (viewfinder only, no face detection) and are shown in the following chart.
+
+![Package Power Consumption](images/face-detection-ptat-fd15fps-rel.png)
+
+Javacript test programs were created to capture frames and to detect faces at VGA resolution (640x480) at 15 fps. The tests were run on Intel Tigerlake running Windows 11. A test run length was 120 seconds (2 minutes) with 640x480 pixel frame resolution. 
+
+## User research
+
+* TEAMS : Supportive, but [not supportive of adding facial expressions](https://www.w3.org/2021/11/24-webrtc-minutes.html#t04) and doubtful on the accuracy of emotion analysis.
+
+*Agreed, Facial expressions are not part of [Non-goals](https://github.com/riju/faceDetection/blob/main/explainer.md#non-goals) for this API*
+
+
+* MEET : Supportive, but many applications might not be using rectangle-bounding box, mask/contour more useful.
+
+*Currently common platforms such as ChromeOS, Android, and Windows support [system APIs](https://github.com/riju/faceDetection/blob/main/explainer.md#platform-support) which return only face bounding box and landmarks, not accurate contour, and therefore initial implementations are expected to support only bounding-box (ie. a contour with maximum of four points). We would keep the API extensible, so that proper contour support can be added in future. A few [use-cases for bounding-box face detection](https://github.com/riju/faceDetection/blob/main/explainer.md#key-scenarios) are listed.*
+
+
+* Zoom : 
+
+## Face Detection API
+
+### Common API
+
+```js
+dictionary DetectedFace {
+  required long                     id;
+  required float                    probability;
+  FrozenArray<Point2D>              contour;
+  FrozenArray<DetectedFaceLandmark> landmarks;
+};
+
+dictionary DetectedFaceLandmark {
+  required FrozenArray<Point2D> contour;
+  FaceLandmark                  type;
+};
+
+enum FaceLandmark {
+  "eye",
+  "eyeLeft",
+  "eyeRight",
+  "mouth",
+  "nose"
+};
+
+partial dictionary MediaTrackSupportedConstraints {
+  boolean faceDetectionMode = true;
+};
+
+partial dictionary MediaTrackCapabilities {
+  sequence<DOMString> faceDetectionMode;
+};
+
+partial dictionary MediaTrackConstraintSet {
+  ConstrainDOMString faceDetectionMode;
+};
+
+partial dictionary MediaTrackSettings {
+  DOMString faceDetectionMode;
+};
+
+enum FaceDetectionMode {
+  "none",         // Face detection is not needed
+  "presence",     // Only the presence of face or faces is returned, not location
+  "bounding-box", // Return bounding box for face
+  "contour",      // Approximate contour of the detected faces is returned
+  "landmarks",    // Approximate contour of the detected faces is returned with facial landmarks

```

Comment:
Do we need all these values?
What is the practical advantage over a simple true/false value?

Reply:
user should be able to select just face detection without landmarks, as it could save computation. Otherwise, removed other options